fdbmonitor/fdbmonitor.cpp

```diff
@@ -1534,21 +1612,33 @@ int main(int argc, char** argv) {
 		}
 
 		double end_time = std::numeric_limits<double>::max();
+		double now = timer();
+		bool need_rss_check = false;
 		for (auto& i : id_command) {
 			if (i.second->fork_retry_time >= 0) {
 				end_time = std::min(i.second->fork_retry_time, end_time);
 			}
+			if (i.second->memory_rss > 0 && id_pid.count(i.first) > 0) {
+				need_rss_check = true;
+			}
+		}
+		bool timeout_for_rss_check = false;
+		if (need_rss_check && end_time > now + RSS_CHECK_INTERVAL) {
```

Comment:
```suggestion
		// If resident memory checks are needed and the planned wake time is longer than the rss check
		// interval from now then shorten it and indicate the end time was set by the need for an rss check. 
		if (need_rss_check && end_time > now + RSS_CHECK_INTERVAL) {
```

Reply:
I think the logic as written resets the wait time until the RSS check any time the loop begins, as the check will always happen `RSS_CHECK_INTERVAL` after `now`.  I think this means that if another process is due to be restarted in, say, 0.7 second from now (the `fork_retry_time`), then the `end_time` will be 0.7 seconds from now, so this loop will not do the rss check, and the next loop might but only if nothing else is scheduled to happen before `RSS_CHECK_INTERVAL`.  Similar, process deaths will halt the sleep and since that isn't a timeout based wakeup the rss check would not run even if it was scheduled.

So basically if processes are dying and restarting frequently, I think the rss check would be avoided pretty often which could make things worse.

Reply:
If my comment above is correct, then I suggest tracking the last time the rss check was done and using that as a base for when it should run next if required.

Reply:
good catch. will fix.